### PR TITLE
init: fix "waiting for the server to be ready" bug.

### PR DESCRIPTION
**Pull request by danielstuart14**
Waiting for a var that'll never be set again isn't smart.

This fixes a possible hang in the AL init for the first player to join the
mission. (If the server isn't configured to auto initialize)

#### Changes proposed in this pull request: 
- Fix client function that waits 'till the server is ready. (Core Init)

### CODE REVIEW

File: Altis_Life.Altis/core/init.sqf [hunk 1]
```diff
@@ -45,15 +45,14 @@ diag_log "::Life Client:: Waiting for the server to be ready..";
 waitUntil{!isNil "life_HC_isActive"};
 if (life_HC_isActive) then {
     waitUntil{!isNil "life_HC_server_isReady" && !isNil "life_HC_server_extDB_notLoaded"};
-    _server_isReady = life_HC_server_isReady;
+    waitUntil{life_HC_server_isReady};
```
Comment (BoGuu) [inline Altis_Life.Altis/core/init.sqf]: The HC isReady is only true IIRC, so we shouldn't need to await this one, but the other is needed :)

File: Altis_Life.Altis/core/init.sqf [hunk 2]
```diff
@@ -45,15 +45,14 @@ diag_log "::Life Client:: Waiting for the server to be ready..";
...
     _extDB_notLoaded = life_HC_server_extDB_notLoaded;
 } else {
     waitUntil{!isNil "life_server_isReady" && !isNil "life_server_extDB_notLoaded"};
-    _server_isReady = life_server_isReady;
+    waitUntil{life_server_isReady};
```
Comment (Jawshy) [inline Altis_Life.Altis/core/init.sqf]: `waitUntil {life_server_isReady};`* - Whitespace.

File: Altis_Life.Altis/core/init.sqf (hunk 1 above)
Comment (danielstuart14) [inline Altis_Life.Altis/core/init.sqf]: indeed, it's pretty useless as life_HC_isActive does the job. I'll update it.

File: Altis_Life.Altis/core/init.sqf (hunk 2 above)
Comment (danielstuart14) [inline Altis_Life.Altis/core/init.sqf]: I will update the whole file

**Comment by danielstuart14** [comment]
should be ready to merge. :D

File: Altis_Life.Altis/core/init.sqf [hunk 3]
```diff
@@ -42,24 +43,14 @@ diag_log "::Life Client:: Received server functions.";
...
-if (life_HC_isActive) then {
-    waitUntil{!isNil "life_HC_server_isReady" && !isNil "life_HC_server_extDB_notLoaded"};
-    _server_isReady = life_HC_server_isReady;
-    _extDB_notLoaded = life_HC_server_extDB_notLoaded;
-} else {
-    waitUntil{!isNil "life_server_isReady" && !isNil "life_server_extDB_notLoaded"};
-    _server_isReady = life_server_isReady;
-    _extDB_notLoaded = life_server_extDB_notLoaded;
-};
+waitUntil {!isNil "life_server_isReady" && !isNil "life_HC_isActive" && !isNil "life_server_extDB_notLoaded"};
```
Comment (Jawshy) [inline Altis_Life.Altis/core/init.sqf]: Can you do [lazy evaluation](https://community.bistudio.com/wiki/Code_Optimisation) on the latter two conditions?

File: Altis_Life.Altis/core/init.sqf [hunk 4]
```diff
@@ -138,7 +129,7 @@ publicVariableServer "life_fnc_RequestClientId"; //Variable OwnerID for Headless
 
 [] spawn {
     for "_i" from 0 to 1 step 0 do {
-        waitUntil{(!isNull (findDisplay 49)) && (!isNull (findDisplay 602))}; // Check if Inventory and ESC dialogs are open
+        waitUntil {(!isNull (findDisplay 49)) && (!isNull (findDisplay 602))}; // Check if Inventory and ESC dialogs are open
```
Comment (Jawshy) [inline Altis_Life.Altis/core/init.sqf]: Lazy eval?

Comment (danielstuart14) [inline Altis_Life.Altis/core/init.sqf]: can't be done here. It waits for these two displays to be open together.

**Comment by BoGuu** [comment]
LGTM

**Comment by Jawshy** [comment]
LGTM because I did not recieve any errors before entering the server.

**Pull request merged**